### PR TITLE
0224 fix docker entrypoint env overrides

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

File: deploy/docker/README.md [hunk 1]
```diff
@@ -50,48 +50,41 @@ The EMQX broker runs as Linux user `emqx` in the docker container.
...
+The environment variables with `EMQX_` prefix are mapped to key-value pairs in configuration files.
 
 Example:
 
 ```bash
-EMQX_LISTENERS__SSL__DEFAULT__ACCEPTORS <--> listeners.ssl.default.acceptors
-EMQX_ZONES__DEFAULT__MQTT__MAX_PACKET_SIZE <--> zones.default.mqtt.max_packet_size
+EMQX_DASHBOARD__DEFAULT_PASSWORD       <--> dashboard.default_password
+EMQX_NODE__COOKIE                      <--> node.cookie
+EMQX_LISTENERS__SSL__default__ENABLE   <--> listeners.ssl.default.enable
```
Comment (Rory-Z) [inline deploy/docker/README.md:60]:
```suggestion
EMQX_LISTENERS__SSL__DEFAULT__ENABLE   <--> listeners.ssl.default.enable
```

**Review by Rory-Z — COMMENTED** [review]
I think we should add `EMQX_NODE_NAME` to `etc/emqx.conf`, because when the user executes `bin/emqx ctl` in the container, `bin/emqx` may not get the environment variables from `docker-entrypoint.sh`

**Comment by zmstone** [comment]
> I think we should add `EMQX_NODE_NAME` to `etc/emqx.conf`, because when the user executes `bin/emqx ctl` in the container, `bin/emqx` may not get the environment variables from `docker-entrypoint.sh`

The node name is retrieved from parsing the `ps -ef` command output before executing the `ctl` command.

**Comment by zmstone** [comment]
> Looks EMQX 4.4 has same issue

For 4.4 the node name is retrieved from the latest `vm.<time>.args` file.
If the node is never initialized or not running when executing non-boot commands, the command id doomed to fail anyway.

File: deploy/docker/README.md (hunk 1 above)
Comment (zmstone) [reply deploy/docker/README.md:60]: this is intentionally left lower case to demo that lower case letters also work.

Comment (Rory-Z) [reply deploy/docker/README.md:60]: Cool, maybe add some note, it's easy to get confused

**Comment by Rory-Z** [comment]
> > I think we should add `EMQX_NODE_NAME` to `etc/emqx.conf`, because when the user executes `bin/emqx ctl` in the container, `bin/emqx` may not get the environment variables from `docker-entrypoint.sh`
> 
> The node name is retrieved from parsing the `ps -ef` command output before executing the `ctl` command.

When I create a EMQX container like this:  `docker run -d --name emqx -e EMQX_NODE_NAME="emqx@fake" emqx/emqx:5.0.14`
check log, the EMQX looks good:
```
$ docker logs -f emqx
WARNING: Default (insecure) Erlang cookie is in use.
WARNING: Configure node.cookie in /opt/emqx/etc/emqx.conf or override from environment variable EMQX_NODE__COOKIE
WARNING: Use the same config value for all nodes in the cluster.
EMQX_RPC__PORT_DISCOVERY [rpc.port_discovery]: manual
EMQX_LOG__FILE_HANDLERS__DEFAULT__ENABLE [log.file_handlers.default.enable]: false
EMQX_LOG__CONSOLE_HANDLER__ENABLE [log.console_handler.enable]: true
EMQX_NODE__NAME [node.name]: emqx@fake
Listener ssl:default on 0.0.0.0:8883 started.
Listener tcp:default on 0.0.0.0:1883 started.
Listener ws:default on 0.0.0.0:8083 started.
Listener wss:default on 0.0.0.0:8084 started.
Listener http:dashboard on :18083 started.
EMQX 5.0.14 is running now!
```

but `emqx ctl` is not working:
```
$ docker exec -it emqx bash
emqx@73c6ca2d0ba5:/opt/emqx$ emqx ctl status
Node emqx@fake not responding to pings.
ERROR: node_is_not_running!
emqx@73c6ca2d0ba5:/opt/emqx$ emqx_ctl status
Node emqx@fake not responding to pings.
ERROR: node_is_not_running!
```

**Comment by zmstone** [comment]
> emqx@fake
in this example, `emqx ctl` actually managed to find the node name.
but failed to ping the node because `fake` is not a valid hostname.

**Review by Rory-Z — APPROVED** [review]

**Pull request merged**